sdk/src/network-client.ts

```diff
@@ -137,12 +137,19 @@ class AleoNetworkClient {
      * @param url The URL to fetch data from.
      */
     async fetchData<Type>(url = "/"): Promise<Type> {
-        try {
-            return parseJSON(await this.fetchRaw(url));
-        } catch (error) {
-            throw new Error(`Error fetching data: ${error}`);
-        }
-    }
+      try {
+          return await retryWithBackoff(() => this.fetchRaw(url).then(parseJSON), {
+              retryOnStatus: [500, 502, 503, 504],
+              shouldRetry: (err) => {
+                  const msg = err?.message?.toLowerCase?.() || "";
+                  return msg.includes("network") || msg.includes("timeout") || msg.includes("503");
```

Comment:
```suggestion
                  return msg.includes("network") || msg.includes("timeout") || msg.includes("503");
```

Why these 3 things in particular?

Reply:
My intention here was to catch common transient network errors, but I'm definitely happy to refine if you have other error responses in mind!

Reply:
I updated the helper to always retry on 5XXs and only apply `shouldRetry` for 4XXs, so `fetchData` now just uses the default behavior